src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -542,9 +545,19 @@ private function parseDefinition(string $id, array|string|null $service, string
         }
 
         if (isset($service['configurator'])) {
-            $definition->setConfigurator($this->parseCallable($service['configurator'], 'configurator', $id, $file));
+            $definition->addConfigurator($this->parseCallable($service['configurator'], 'configurator', $id, $file));
```

Comment:
Have you added a depreciation message when this attribute is used? I think we should: having both `configurator` and `configurators` is confusing.

Not needed for other configuration formats as the name don't have to be pluralized in XML or PHP.